src/sardana/spock/spockms.py

```diff
@@ -38,19 +38,21 @@
 from sardana.sardanautils import is_pure_str, is_non_str_seq
 from sardana.spock import genutils
 from sardana.util.parser import ParamParser
-from sardana.spock.inputhandler import SpockInputHandler, InputHandler
 from sardana import sardanacustomsettings
 
 CHANGE_EVTS = TaurusEventType.Change, TaurusEventType.Periodic
 
 
 if genutils.get_gui_mode() == 'qt':
+    from taurus.external.qt import Qt
     from sardana.taurus.qt.qtcore.tango.sardana.macroserver import QDoor, QMacroServer
+    from sardana.spock.qtinputhandler import InputHandler
     BaseDoor = QDoor
     BaseMacroServer = QMacroServer
     BaseGUIViewer = object
 else:
     from sardana.taurus.core.tango.sardana.macroserver import BaseDoor, BaseMacroServer
+    from sardana.spock.inputhandler import SpockInputHandler
```

Comment:
We can not import it only here. The default behavior (see `sardanacustomsettings.SPOCK_INPUT_HANDLER` uses the `SpockInputHandler` even if the Qt is installed. So, `genutils.get_gui_mode()` is not enough to determine which input handler to import. I propose to move both imports to the `create_input_handler()` methods of the _doors_.

Reply:
From my code analysis, current solution does not use `SpockInputHandler` at all because of two reasons:
 - `genutils.get_gui_mode()` always returns `qt` which makes `if` condition redundant
 - there was an error in when instantiating `SpockInputHanlder` in `create_input_hanlder` method. In current version, there was no way to get it to work AFAIK.

This was my analysis - I may be wrong, correct me please if I am.

I'm not entirely sure that I know what you mean so I'll push changes and you will be able to see if that's what you meant.

Reply:
Regarding imports - only import of input handlers, right?

Reply:
> From my code analysis, current solution does not use `SpockInputHandler` at all because of two reasons:

The `SpockInputHandler` was instantiated correctly in the [`QSpockDoor.create_input_handler()`](https://github.com/sardana-org/sardana/blob/f2f73ac20b1f397450109a582e264577cfabd011/src/sardana/spock/spockms.py#L584).
This is a typical scenario - Qt is installed and `sardanacustomsetting.SPOCK_INPUT_HANDLER = "CLI".

You are right that the creation on `SpockInputHandler` in the [`SpockBaseDoor.create_input_handler()`](https://github.com/sardana-org/sardana/blob/f2f73ac20b1f397450109a582e264577cfabd011/src/sardana/spock/spockms.py#L293) was erroneus and you fixed it. But this scenario was rather not used.

Reply:
Okay, right, I see it now.